pkg/validation/internal/operatorhub_test.go

```diff
@@ -129,46 +130,331 @@ func TestExtractCategories(t *testing.T) {
 	}
 }
 
-func TestWarnNoIcon(t *testing.T) {
-	var table = []struct {
-		description string
-		directory   string
-		hasError    bool
+func TestCheckSpecIcon(t *testing.T) {
+	validIcon := v1alpha1.Icon{
+		Data:      "iVBORw0KGgoAAAANSUhEUgAAAOEAAADZCAYAAADWmle6AAAACXBIWXMAAAsTAAALEwEAmpwYAAAAGXRFWHRTb2Z0d2Fy",
+		MediaType: "image/png",
+	}
+	invalidIcon := v1alpha1.Icon{
+		MediaType: "image/png",
+	}
+
+	invalidMediaTypeIcon := validIcon
+	invalidMediaTypeIcon.MediaType = "invalid"
+
+	type args struct {
+		icon []v1alpha1.Icon
+	}
+	tests := []struct {
+		name        string
+		args        args
+		wantError   bool
+		wantWarning bool
 		errStrings  []string
 		warnStrings []string
 	}{
 		{
-			description: "valid bundle no icon",
-			directory:   "./testdata/valid_bundle_no_icon",
```

Comment:
we do not add and read a bundle. we can just mock the data to be able to cover all scenarios. 
See that we are now covering all issues/errors that this check can raise instead of only checking only one scenario as before.